src/core/prompts.ts

```diff
@@ -1,6 +1,10 @@
 export const createCodingPrompt = (
-  originalPrompt: string
-): string => `You are a senior software engineer.
+  originalPrompt: string,
+  customPrompt?: string
+): string => {
+  const basePrompt =
+    customPrompt ||
```

Comment:
duckling instead of treating custom prompt and this prompt as separate, the prompt below should be the default and the prompt passed through is always the setting one